graylog2-web-interface/src/views/logic/slices/viewSlice.ts

```diff
@@ -122,7 +123,38 @@ export const loadView = (newView: View, recreateSearch: boolean = false) => asyn
 };
 
 export const updateView = (newView: View, recreateSearch: boolean = false) => async (dispatch: AppDispatch, getState: () => RootState) => {
-  const view = selectView(getState());
+  const state = getState();
+  const view = selectView(state);
+
+  await dispatch(pushIntoBuffer({
+    type: 'view',
+    state: {
+      ...state.view,
+    },
+  }));
+
+  if (recreateSearch || !isViewEqualForSearch(view, newView)) {
+    const updatedViewWithSearch = await _recreateSearch(newView);
+    await dispatch(setView(updatedViewWithSearch, true));
+
+    return dispatch(execute());
+  }
+
+  return dispatch(setView(newView, true));
+};
+
+export const updateViewNoBufferPush = ({ newView, recreateSearch = false, hasToPushRevision = false }: { newView: View, recreateSearch: boolean, hasToPushRevision: boolean }) => async (dispatch: AppDispatch, getState: () => RootState) => {
```

Comment:
This is essentially the same as `updateView`. Maybe we should add the `pushRevision` argument to `updateView` and delete `updateViewNoBufferPush`?